connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -0,0 +1,195 @@
+            connect.stop();
+        } catch (Exception e) {
+            log.error("Could not stop connect", e);
+        }
+
+        try {
+            kafkaCluster.after();
+        } catch (Exception e) {
+            log.error("Could not stop kafka", e);
+        }
+    }
+
+    public void startConnect() {
+        log.info("Starting Connect cluster with one worker.");
+
+        workerProps.put(BOOTSTRAP_SERVERS_CONFIG, kafka().bootstrapServers());
+        workerProps.put(REST_HOST_NAME_CONFIG, REST_HOST_NAME);
+
+        putIfAbsent(workerProps, GROUP_ID_CONFIG, "connect-integration-test-" + clusterName);
+        putIfAbsent(workerProps, OFFSET_STORAGE_TOPIC_CONFIG, "connect-offset-topic-" + clusterName);
+        putIfAbsent(workerProps, OFFSET_STORAGE_REPLICATION_FACTOR_CONFIG, "1");
+        putIfAbsent(workerProps, CONFIG_TOPIC_CONFIG, "connect-config-topic-" + clusterName);
+        putIfAbsent(workerProps, CONFIG_STORAGE_REPLICATION_FACTOR_CONFIG, "1");
+        putIfAbsent(workerProps, STATUS_STORAGE_TOPIC_CONFIG, "connect-storage-topic-" + clusterName);
+        putIfAbsent(workerProps, STATUS_STORAGE_REPLICATION_FACTOR_CONFIG, "1");
+        putIfAbsent(workerProps, KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
+        putIfAbsent(workerProps, VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
+
+        connect = new ConnectDistributed().startConnect(workerProps);
+    }
+
+    public void startConnector(String connName, Map<String, String> connConfig) throws IOException {
+        String url = String.format("http://%s:%s/connectors/%s/config", REST_HOST_NAME, WorkerConfig.REST_PORT_DEFAULT, connName);
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            String content = mapper.writeValueAsString(connConfig);
+            int status = executePut(url, content);
+            if (status >= HttpServletResponse.SC_BAD_REQUEST) {
+                throw new IOException("Could not execute PUT request. status=" + status);
+            }
+        } catch (IOException e) {
+            log.error("Could not serialize config", e);
+            throw new IOException(e);
+        }
+    }
+
+    public void deleteConnector(String connName) throws IOException {
+        int status = executeDelete(String.format("http://%s:%s/connectors/%s", REST_HOST_NAME, WorkerConfig.REST_PORT_DEFAULT, connName));
+        if (status >= HttpServletResponse.SC_BAD_REQUEST) {
+            throw new IOException("Could not execute DELETE request. status=" + status);
+        }
+    }
+
+    private static void putIfAbsent(Map<String, String> props, String propertyKey, String propertyValue) {
+        if (!props.containsKey(propertyKey)) {
+            props.put(propertyKey, propertyValue);
+        }
+    }
+
+    public EmbeddedKafkaCluster kafka() {
+        return kafkaCluster;
+    }
+
+    public int executePut(String url, String body) throws IOException {
```

Comment:
How about one for POST?

Reply:
do we need one right now? most of the connect API uses GET or PUT. POST is only for restarting connectors or tasks.